rotkehlchen/api/v1/schemas.py

```diff
@@ -720,49 +725,168 @@ def generate_fields_post_validation(self, data: dict[str, Any]) -> dict[str, Any
+        validator_index = fields.Integer(
+            required=True,
+            validate=webargs.validate.Range(
+                min=0,
+                error='Validator index must be an integer >= 0',
+            ),
+        )
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            return {'event': EthBlockEvent(**data)}
+
+    class CreateEthDepositEventEventSchema(BaseSchema):
+        tx_hash = EVMTransactionHashField(required=True)
+        depositor = EvmAddressField(required=True)
+        event_identifier = fields.String(required=True)
+        sequence_index = fields.Integer(required=True)
+        validator_index = fields.Integer(
+            required=True,
+            validate=webargs.validate.Range(
+                min=0,
+                error='Validator index must be an integer >= 0',
+            ),
+        )
+        extra_data = fields.Dict(load_default=None)
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            return {'event': EthDepositEvent(**data)}
+
+    class CreateEthWithdrawalEventEventSchema(BaseSchema):
+        is_exit = fields.Boolean(required=True)
+        withdrawal_address = EvmAddressField(required=True)
+        validator_index = fields.Integer(
+            required=True,
+            validate=webargs.validate.Range(
+                min=0,
+                error='Validator index must be an integer >= 0',
+            ),
+        )
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            return {'event': EthWithdrawalEvent(**data)}
 
     @post_load
     def make_history_base_entry(
             self,
             data: dict[str, Any],
             **_kwargs: Any,
     ) -> dict[str, Any]:
-        if data.get('event_subtype', None) is None:
-            data['event_subtype'] = HistoryEventSubType.NONE
-
-        if data['location'] not in EVM_LOCATIONS:
-            raise ValidationError(
-                message=f'EVM event location needs to be one of {EVM_LOCATIONS}',
-                field_name='location',
-            )
+        entry_type = data.pop('entry_type')
+        entry_to_schema: dict[HistoryBaseEntryType, type[Schema]] = {
+            HistoryBaseEntryType.HISTORY_EVENT: self.CreateBaseHistoryEventSchema,
+            HistoryBaseEntryType.ETH_BLOCK_EVENT: self.CreateEthBlockEventEventSchema,
+            HistoryBaseEntryType.ETH_DEPOSIT_EVENT: self.CreateEthDepositEventEventSchema,
+            HistoryBaseEntryType.ETH_WITHDRAWAL_EVENT: self.CreateEthWithdrawalEventEventSchema,
+            HistoryBaseEntryType.EVM_EVENT: self.CreateEvmEventSchema,
+        }
```

Comment:
maybe define this mapping as a class attribute? Otherwise you are creating it every time the endpoint is executed